yarn-project/archiver/src/archiver/archiver_store.ts

```diff
@@ -347,6 +361,60 @@ export class MemoryArchiverStore implements ArchiverDataStore {
     return Promise.resolve(logs.slice(startIndex, endIndex));
   }
 
+  /**
+   * Gets unencrypted logs based on the provided filter.
+   * @param filter - The filter to apply to the logs.
+   * @returns The requested logs.
+   * @throws If txHash and block range are both defined.
+   */
+  getUnencryptedLogs(filter: LogFilter): Promise<ExtendedUnencryptedL2Log[]> {
+    if (filter.txHash && (filter.fromBlock || filter.toBlock)) {
+      throw new Error('Cannot filter by txHash and block range at the same time');
+    }
+
+    const txHash = filter.txHash;
+
+    const fromBlockIndex = filter.fromBlock === undefined ? 0 : filter.fromBlock - INITIAL_L2_BLOCK_NUM;
+    if (fromBlockIndex < 0) {
+      throw new Error(`"fromBlock" (${filter.fromBlock}) smaller than genesis block number (${INITIAL_L2_BLOCK_NUM}).`);
+    }
+    if (fromBlockIndex > this.unencryptedLogs.length) {
```

Comment:
Should we rename `unencryptedLogs` to `unencryptedLogsAtBlock` or `unencryptedLogsPerBlock`? This line confused me for a while...

Can be left for another PR, of course.

Reply:
Sneaked it in [this commit](https://github.com/AztecProtocol/aztec-packages/pull/2600/commits/dcc234d92aa144badebe85c27f9860824805e7b2)